lib/conjur/base.rb

```diff
@@ -169,6 +169,13 @@ def initialize username, api_key, token, remote_ip = nil
       @username = username
       @api_key = api_key
       @token = token
+      
+      # Assume that the token was recently minted, log a warning if not
+      @token_born = gettime   
+      if token && (delta = Time.now - Time.parse(token['timestamp'])) > 10
+        $stderr.puts "WARNING: token timestamp offset by #{delta.to_i} seconds"
+      end
+
```

Comment:
A monotonic clock is more robust for measuring the passage of time. RTC
isn't really appropriate for that: due to time synchronization and leap
seconds it can jump back and forth, slow down or speed up.
22 cze 2016 13:02 "apotterri" <notifications@github.com> napisał(a):

> In lib/conjur/base.rb
> <https://github.com/conjurinc/api-ruby/pull/77#discussion_r68033921>:
>
> > @@ -169,6 +169,13 @@ def initialize username, api_key, token, remote_ip = nil
> >        @username = username
> >        @api_key = api_key
> >        @token = token
> > +
> > +      # Assume that the token was recently minted, log a warning if not
> > +      @token_born = gettime
> > +      if token && (delta = Time.now - Time.parse(token['timestamp'])) > 10
> > +        $stderr.puts "WARNING: token timestamp offset by #{delta.to_i} seconds"
> > +      end
> > +
>
> Seems like a simpler alternative might be to always use Time.now. What
> was the reason for using CLOCK_MONOTONIC?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/conjurinc/api-ruby/pull/77/files/6afb0002a561235aebb23f51414863c60eeccd98#r68033921>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAyRVM-U5hMbYaI_0rQGMrlWilGMjPoAks5qORYtgaJpZM4I7PQS>
> .
>